src/mc_rbdyn/Robot.cpp

```diff
@@ -524,6 +524,19 @@ const RobotModule & Robot::module() const
   return robots_->robotModule(robots_idx_);
 }
 
+void Robot::addBodySensor(const BodySensor & sensor) const
+{
+  if(!hasBodySensor(sensor.name()))
+  {
+    data_->bodySensors.push_back(sensor);
+    data_->bodySensorsIndex.insert({sensor.name(), data_->bodySensors.size()});
+
+    if(!bodyHasBodySensor(sensor.name())) data_->bodyBodySensors.insert({sensor.name(), data_->bodySensors.size()});
```

Comment:
```suggestion
    if(!bodyHasBodySensor(sensor.name())) data_->bodyBodySensors.insert({sensor.name(), data_->bodySensors.size() -1});
```

Otherwise this is an invalid index